src/edges_analysis/filters/filters.py

```diff
@@ -817,8 +818,48 @@ def object_rms_filter(
         rms = np.sqrt(np.mean(data.residuals**2))
 
     logger.info(f"RMS for {data.name}: {rms:.2f} mK")
-
     return GSFlag(
         flags=np.array([rms > rms_threshold]),
         axes=("time",),
     )
+
+
+@gsregister("filter")
+@gsdata_filter()
+def explicit_day_filter(
+    data: GSData,
+    flag_days: list[tuple[int, int] | tuple[int, int, int] | int | Time],
+) -> GSFlag:
+    """Filter out any data coming from specific days.
+
+    Parameters
+    ----------
+    flag_days
+        A list of days to flag. Each entry can be a 2-tuple, 3-tuple, astropy.Time or an
+        int. If a 2-tuple, it is interpreted as ``(year, day_of_year)``. If a 3-tuple,
+        it is interpreted as ``(year, month, day)``. If an int, it is interpreted as a
+        Julian day.
+    """
+    day_flags = np.zeros_like(flag_days)
+    for i, day in enumerate(flag_days):
+        if hasattr(day, "__len__"):
+            if len(day) == 2:
+                t = Time(f"{day[0]:04}:{day[1]:03}:00:00:00.000", format="yday")
+            elif len(day) == 3:
+                t = Time(
+                    f"{day[0]:04}-{day[1]:02}-{day[2]:02} 00:00:00.000", format="iso"
+                )
+            else:
+                raise ValueError("Day must be a 2-tuple, 3-tuple, Time or an int.")
+
+            day_flags[i] = int(t.jd)  # t.value
+        elif isinstance(day, Time):
+            flag_days[i] = int(day.jd)
+
+    thist = Time(f"{data.name.replace('_',':')}:00:00:00.000", format="yday")
```

Comment:
Using the `data.name` attribute was necessary previously because once we were in "lst mode" we removed the actual times. However, this is not the case any more in `GSData`. This means we can be more careful here, and do something like

```python
thist = data.times[:, 0].jd.astype(int)
flags = np.isin(thist, flag_days)
return GSFlag(flags=dlags, axes=("time",))
```

Reply:
@steven-murray done. Can we merge it?